test/service.spec.js

```diff
@@ -1,16 +1,48 @@
 // Generated by CoffeeScript 1.12.3
 (function() {
-  var AWS, MockService, helpers, metadata,
+  var AWS, MockService, helpers, metadata, MockServiceWithErrorCodeMapping
     hasProp = {}.hasOwnProperty;
 
   helpers = require('./helpers');
 
   AWS = helpers.AWS;
 
   MockService = helpers.MockService;
+  MockServiceWithErrorCodeMapping = helpers.MockServiceWithErrorCodeMapping;
 
   metadata = require('../apis/metadata.json');
 
+  describe('AWSQuery compatible error codes', function () {
+    var service;
+    var mapping = { ServiceError: { code: 'Custom.ErrorCode' } };
```

Comment:
to make it more obvious what's being tested, could we have test data that's obviously test data so that it's less likely to be confused with something else? e.g.

`{ AnArbitraryResponseError: { code: 'AMappedErrorCode' } }`

Reply:
This is a bit harder to implement due to the way MockService is implemented. I think the tests give us confidence that the newly introduced attributes work fine. So, I will keep it this way for now. Will consider adding SQS specific tests if time permits.